src/abi-cache/redis-abi-cache.ts

```diff
@@ -0,0 +1,42 @@
+import { Redis, RedisOptions } from 'ioredis'
+
+import { stringify } from '../lib/stringify'
+
+import { AbiCache, AbiInformation } from './abi-cache'
+
+export class RedisAbiCache extends AbiCache {
+  private redis: Redis
+  private cachePrefix = 'ABICACHE:'
+
+  constructor(redisOptions: RedisOptions) {
+    super()
+    this.redis = new Redis({ ...redisOptions, keyPrefix: this.cachePrefix })
+  }
+
+  public override async init(): Promise<AbiCache> {
+    const keys = await this.redis.keys(`${this.cachePrefix}*`)
+
+    await Promise.all(
+      keys.map(async (key) => {
+        console.log(`Reading ABI ${key}`)
+        const abiKey = key.replace(`${this.cachePrefix}`, '')
+        const abi = await this.redis.get(abiKey)
+
+        if (abi) {
+          this.loadFromString(abiKey, abi)
+        }
+      })
+    )
+
+    return Promise.resolve(this)
+  }
+
+  protected override async persist(
+    key: string,
+    abi: AbiInformation
+  ): Promise<void> {
+    console.log(`Persisting ABI ${key}`)
```

Comment:
We should not log anything with `console.log` since this will be emitted on our backend pods then and since it's not a valid JSON format in the ECS format that we use for ES, it will not be parseable by our ES tool chain.
Either we allow to pass a logger (that's how we do it on the rpc wrapper I think) or we just don't log I would say

Reply:
https://github.com/lifinance/abi-parser/pull/62